iOS/Flows/Conversation/Cells/TimeMachineCollectionViewLayout.swift

```diff
@@ -80,17 +89,39 @@ class TimeMachineCollectionViewLayout: UICollectionViewLayout {
         return true
     }
 
-    override func invalidateLayout() {
-        super.invalidateLayout()
+    override func invalidationContext(forBoundsChange newBounds: CGRect)
+    -> UICollectionViewLayoutInvalidationContext {
+
+        let invalidationContext = super.invalidationContext(forBoundsChange: newBounds)
+
+        guard let customInvalidationContext = invalidationContext as? TimeMachineCollectionViewLayoutInvalidationContext else {
+            return invalidationContext
+        }
+
+        // Changing the bounds doesn't affect item z ranges.
+        customInvalidationContext.shouldRecalculateZRanges = false
+
+        return customInvalidationContext
+    }
+
+    override func invalidateLayout(with context: UICollectionViewLayoutInvalidationContext) {
+        super.invalidateLayout(with: context)
+
+        guard let customContext = context as? TimeMachineCollectionViewLayoutInvalidationContext else { return }
 
         // Clear the layout attributes caches.
         self.cellLayoutAttributes.removeAll()
-        self.zRangesDict.removeAll()
         self.decorationLayoutAttributes.removeAll()
+        if customContext.shouldRecalculateZRanges {
+            self.zRangesDict.removeAll()
+        }
```

Comment:
Now, when the layout is invalidated, we check to see if we should delete the z ranges cache.